pkg/planner/core/operator/logicalop/logical_join.go

```diff
@@ -748,7 +637,21 @@ func (p *LogicalJoin) ExtractFD() *funcdep.FDSet {
 	}
 }
 
-// GetBaseLogicalPlan inherits the BaseLogicalPlan.LogicalPlan.<23th> implementation.
+// Keep predicates that *involve at least one column* from the given schema.
+// to belong to `sch`; it just checks “touches inner side or not”.
+func filterPredsInvolvingSchema(preds []expression.Expression, sch *expression.Schema) []expression.Expression {
+	out := make([]expression.Expression, 0, len(preds))
+	for _, e := range preds {
+		cols := expression.ExtractColumns(e)
```

Comment:
We can use `ExtractColumnsMapFromExpressionsWithReusedMap`. it can reuse the memory and remove the same columns.

Reply:
```ExtractColumnsMapFromExpressionsWithReusedMap``` only supplies the filter condition. We simplified the code.